src/main/resources/entity-types/circulation/composite_loan_details.json5

```diff
@@ -14,7 +14,7 @@
       id: '64d7b5fb-2ead-444c-a9bd-b9db831f4132',
       join: {
         type: 'left join',
-        joinTo: 'loans',
+        joinTo: 'loans.loan',
```

Comment:
Afaik, there are only 2 breaking changes from this PR:
- the one here, where `joinTo` has to refer to a specific source in the target entity type (I could probably fix this and maintain the old behavior, but I honestly kind of prefer this approach, since it gives a bit more power when joining)
- In some cases, field names can change. I don't think this actually happened with any of our existing entity types, but it does show up in `EntityTypeFlatteningServiceTest` with the test that flattens a composite containing a composite